File: app/src/main/java/com/urban/mvvmshowcase/view/activity/AbstractSavingStateViewModelActivity.java
package com.urban.mvvmshowcase.view.activity;

import android.os.Bundle;

import com.urban.mvvmshowcase.view.vmwrapper.SavingStateViewModel;

public abstract class AbstractSavingStateViewModelActivity<T extends SavingStateViewModel>
        extends AbstractViewModelActivity<T> {

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        vm().onSaveState(outState);
    }

    @Override
    protected void onRestoreInstanceState(Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);
        vm().onRestoreState(savedInstanceState);
    }
}

File: app/src/main/java/com/urban/mvvmshowcase/view/activity/base/AbstractSavingStateViewModelActivity.kt
package com.urban.mvvmshowcase.view.activity.base

import android.os.Bundle
import com.urban.mvvmshowcase.view.vmwrapper.SavingStateViewModel

abstract class AbstractSavingStateViewModelActivity<out T : SavingStateViewModel>
    : AbstractViewModelActivity<T>() {
    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        vm().onSaveState(outState)
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
        vm().onRestoreState(savedInstanceState)
    }
}